File: java/com/epicdragonworld/gameserver/model/actor/Monster.java
package com.epicdragonworld.gameserver.model.actor;

import com.epicdragonworld.gameserver.model.holders.NpcHolder;
import com.epicdragonworld.gameserver.model.holders.SpawnHolder;

/**
 * @author Pantelis Andrianakis
 */
public class Monster extends Npc
{
	public Monster(NpcHolder npcHolder, SpawnHolder spawnHolder)
	{
		super(npcHolder, spawnHolder);
		
		// TODO: AI Tasks.
		// TODO: Loot corpse.
	}
	
	@Override
	public boolean isMonster()
	{
		return true;
	}
	
	@Override
	public Monster asMonster()
	{
		return this;
	}
}

File: EpicDragonWorldServer/Actor/Monster.cs
/**
 * Author: Pantelis Andrianakis
 * Date: November 28th 2019
 */
public class Monster : Npc
{
    public Monster(NpcHolder npcHolder, SpawnHolder spawnHolder) : base(npcHolder, spawnHolder)
    {
    }

    public override bool IsMonster()
    {
        return true;
    }

    public override Monster AsMonster()
    {
        return this;
    }

    public override string ToString()
    {
        return "MONSTER [" + GetNpcHolder().GetNpcId() + "]";
    }
}
